sdk/test-utils/perf/src/perfTestBase.ts

```diff
@@ -0,0 +1,102 @@
+ * Defines the behavior of the PerfTest constructor, to use the class as a value.
+ */
+export interface PerfTestConstructor<
+  TOptions extends Record<string, unknown> = Record<string, unknown>
+> {
+  new (): PerfTestBase<TOptions>;
+}
+
+/**
+ * Conveys the structure of any Perf test.
+ * It allows developers to define the optional parameters to receive
+ * in a way that aids them to auto-complete these options while adding code inside of each method.
+ * It provides methods to parse the assigned options, as well as to set up and tear down any resources
+ * at a global level (only one setup and teardown for the whole execution of a Perf test),
+ * and at a local level, which happens once for each initialization of the test class
+ * (initializations are as many as the "parallel" command line parameter specifies).
+ */
+export abstract class PerfTestBase<TOptions = Record<string, unknown>> {
+  public abstract options: PerfOptionDictionary<TOptions>;
+
+  public get parsedOptions(): ParsedPerfOptions<TOptions & DefaultPerfOptions> {
+    // Only validate the options if they are defined: if (when) parsedOptions is called
+    // in the constructor, options will be undefined.
+    if (this.options) {
+      validateOptions({
+        ...this.options,
+        ...defaultPerfOptions
+      });
+    }
+
+    // This cast is needed because TS thinks
+    //   PerfOptionDictionary<TOptions & DefaultPerfOptions>
+    //   is different from
+    //   PerfOptionDictionary<TOptions> & PerfOptionDictionary<DefaultPerfOptions>
+    return parsePerfOption({
+      ...this.options,
+      ...defaultPerfOptions
+    } as PerfOptionDictionary<TOptions & DefaultPerfOptions>);
+  }
+
+  private static globalParallelIndex = 0;
+  protected readonly parallelIndex: number;
+
+  public constructor() {
+    this.parallelIndex = PerfTestBase.globalParallelIndex;
+    PerfTestBase.globalParallelIndex++;
+  }
+
+  // Before and after running a bunch of the same test.
+  public globalSetup?(): void | Promise<void>;
+  public globalCleanup?(): void | Promise<void>;
+
+  public postSetup?(): void | Promise<void>;
+  public preCleanup?(): void | Promise<void>;
+
+  public setup?(): void | Promise<void>;
+  public cleanup?(): void | Promise<void>;
+
+  public abstract runAll(
+    parallel: PerfParallel,
+    durationMilliseconds: number,
+    abortController: AbortController
+  ): Promise<void>;
+}
+
+/**
+ * Picks a specific test case by comparing the first command line paramter to the names of the
+ * given classes, all of which must extend PerfTest.
+ * @param tests An array of classes that extend PerfTest
+ */
+export function selectPerfTest(tests: PerfTestConstructor[]): PerfTestConstructor {
+  const testsNames: string[] = tests.map((test) => test.name);
+  const minimistResult: MinimistParsedArgs = minimist(process.argv);
+  const testName = minimistResult._[minimistResult._.length - 1];
+
+  const testIndex = testsNames.indexOf(testName);
+  if (testIndex === -1) {
```

Comment:
```suggestion
  if (testsNames.indexOf(testName) === -1) {
```

I prefer this change, but feel free to ignore this suggestion.

Reply:
`testIndex` is used below to return, I'll keep this.